### PR TITLE
MAIN-8063: Most of the features button in source editor are not present on preview

### PR DESCRIPTION
**Pull request by kerriamber**
This reverts SUS-905

### CODE REVIEW

File: extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php [hunk 1]
```diff
@@ -133,32 +133,30 @@ function SiteWideMessagesSiteNoticeAfter( &$siteNotice ) {
  *
  * @author macbre
  */
-function SiteWideMessagesAddNotifications( Skin $skin, &$tpl ) {
-	global $wgExtensionsPath;
+function SiteWideMessagesAddNotifications(&$skim, &$tpl) {
```
Comment (WikiaTech) [inline extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php:136]: MAJOR Remove the unused function parameters "$skim, $tpl". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

File: extensions/wikia/FacebookClient/FacebookClientHooks.class.php [hunk 2]
```diff
@@ -154,13 +154,17 @@ public static function onSkinTemplatePageBeforeUserMsg( &$html ) {
...
 	 * @param Skin $skin
-	 * @return bool true to continue hook processing
+	 * @param string $text
+	 *
+	 * @return bool
+	 * @throws WikiaException
 	 */
-	public static function onBeforePageDisplay( OutputPage $out, Skin $skin ) {
-		Wikia::addAssetsToOutput( 'facebook_client_fbtags_js' );
+	public static function onSkinAfterBottomScripts( $skin, &$text ) {
```
Comment (WikiaTech) [inline extensions/wikia/FacebookClient/FacebookClientHooks.class.php:163]: MAJOR Remove the unused function parameter "$skin". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

File: skins/oasis/modules/OasisController.class.php [hunk 3]
```diff
@@ -93,22 +94,23 @@ public static function onMakeGlobalVariablesScript(Array &$vars) {
...
+		elseif ($wgTitle->getNamespace() == NS_SPECIAL || BodyController::isEditPage()) {
+			$jsAtBottom = false;
+		}
+		else {
+			$jsAtBottom = true;
+		}
+		return $jsAtBottom;
 	}
 
 	public function executeIndex($params) {
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:112]: MAJOR This function "executeIndex" has 159 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)

File: skins/oasis/modules/OasisController.class.php [hunk 4]
```diff
@@ -292,28 +371,48 @@ private function loadJs() {
...
+		// generate direct script tags
+		foreach ($assets as $url) {
+			$url = htmlspecialchars( $url );
+			$jsLoader .= "<script src=\"{$url}\"></script>\n";
+		}
+
+		$tpl = $this->app->getSkinTemplateObj();
+
+		// $tpl->set( 'headscripts', $out->getHeadScripts() . $out->getHeadItems() );
+		// FIXME: we need to remove head items - i.e. <meta> tags
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:400]: MAJOR Take the required action to fix the issue indicated by this "FIXME" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)

File: skins/oasis/modules/OasisController.class.php [hunk 5]
```diff
@@ -93,22 +94,23 @@ public static function onMakeGlobalVariablesScript(Array &$vars) {
...
+		}
+		else {
+			$jsAtBottom = true;
+		}
+		return $jsAtBottom;
 	}
 
 	public function executeIndex($params) {
-		global $wgOut, $wgUser, $wgRequest, $wgOasisThemeSettings,
+		global $wgOut, $wgUser, $wgTitle, $wgRequest, $wgEnableAdminDashboardExt, $wgOasisThemeSettings,
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:113]: MAJOR Remove this unused "$wgTitle" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)
MAJOR Remove this unused "$wgEnableAdminDashboardExt" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

File: skins/oasis/modules/OasisController.class.php [hunk 6]
```diff
@@ -292,28 +371,48 @@ private function loadJs() {
...
 
-		// SUS-905: Load all AssetsManager JS groups in a single HTTP request
-		/** @var SkinOasis $skin */
-		$skin = $this->getContext()->getSkin();
-		$jsLoader = $skin->getScriptsWithCombinedGroups( $assetGroups );
+		$assets = $this->assetsManager->getURL( $assetGroups ) ;
+
+		// jQueryless version - appears only to be used by the ad-experiment at the moment.
+		// disabled - not needed atm (and skipped in wsl-version anyway)
+		// $assets[] = $this->assetsManager->getURL( $isLoggedIn ? 'oasis_nojquery_shared_js_user' : 'oasis_nojquery_shared_js_anon' );
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:386]: MAJOR Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)

File: skins/oasis/modules/OasisController.class.php [hunk 7]
```diff
@@ -173,6 +175,9 @@ public function executeIndex($params) {
 		$this->itemType = self::getItemType();
 
 		$skin = RequestContext::getMain()->getSkin(); /* @var $skin WikiaSkin */
+		// this is bad but some extensions could have added some scripts to bottom queue
+		// todo: make it not run twice during each request
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:179]: INFO Complete the task associated to this "TODO" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

File: skins/oasis/modules/OasisController.class.php [hunk 8]
```diff
@@ -264,15 +269,67 @@ public function executeIndex($params) {
...
+							return $this->assetsManager->getOneCommonURL($replacement . $matches[1],true);
+						}
+					}
+				}
+			}
+		}
+		return $url;
+	}
+
 	// TODO: implement as a separate module?
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:323]: INFO Complete the task associated to this "TODO" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1135)

File: skins/oasis/modules/OasisController.class.php [hunk 9]
```diff
@@ -264,15 +269,67 @@ public function executeIndex($params) {
...
 	private function loadJs() {
-		global $wgJsMimeType, $wgUser;
+		global $wgJsMimeType, $wgUser, $wgDevelEnvironment, $wgAllInOne;
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:325]: MAJOR Remove this unused "$wgDevelEnvironment" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

File: skins/oasis/modules/OasisController.class.php [hunk 10]
```diff
@@ -264,15 +269,67 @@ public function executeIndex($params) {
...
 	}
 
+	/**
+	 * Gets the URL and converts it to minified one if it points to single static file (JS or CSS)
+	 * If it's not recognized as static asset the original URL is returned
+	 *
+	 * @param $url string URL to be inspected
+	 * @return string
+	 */
+	private function minifySingleAsset( $url ) {
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:279]: MAJOR Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)

File: skins/oasis/modules/OasisController.class.php [hunk 11]
```diff
@@ -264,15 +269,67 @@ public function executeIndex($params) {
...
+
+				// BugId: 38195 - wgExtensionPath / stylePath / ResourceBasePath do not end with a slash
+				// add one to remove double slashes in resulting URL
+				$prefix .= '/';
+
+				if (startsWith($url, $prefix)) {
+					$nurl = substr($url,strlen($prefix));
+					$matches = array();
+					if (preg_match("/^([^?]+)/",$nurl,$matches)) {
+						if (preg_match("/\\.(css|js)\$/i",$matches[1])) {
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:313]: MAJOR Merge this if statement with the enclosing one. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)
MAJOR Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS134)

File: skins/oasis/modules/OasisController.class.php [hunk 12]
```diff
@@ -93,22 +94,23 @@ public static function onMakeGlobalVariablesScript(Array &$vars) {
...
+		// decide where JS should be placed (only add JS at the top for non-search Special and edit pages)
+		if (WikiaPageType::isSearch() || WikiaPageType::isForum()) {
+			// Remove this whole condition when AdDriver2.js is fully implemented and deployed
+
+			$jsAtBottom = true;
+		}
+		elseif ($wgTitle->getNamespace() == NS_SPECIAL || BodyController::isEditPage()) {
+			$jsAtBottom = false;
+		}
+		else {
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:106]: MAJOR This branch's code block is the same as the block for the branch on line 98. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1871)

File: skins/oasis/modules/OasisController.class.php [hunk 13]
```diff
@@ -282,6 +339,28 @@ private function loadJs() {
 			$this->globalBlockingScripts .= "<script type=\"$wgJsMimeType\" src=\"$blockingFile\"></script>";
 		}
 
+		// move JS files added to OutputPage to list of files to be loaded
+		$scripts = RequestContext::getMain()->getSkin()->getScripts();
+
+			foreach ( $scripts as $s ) {
+			//add inline scripts to jsFiles and move non-inline to the queue
+			if ( !empty( $s['url'] ) ) {
+				// FIXME: quick hack to load MW core JavaScript at the top of the page - really, please fix me!
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:348]: MAJOR Take the required action to fix the issue indicated by this "FIXME" comment. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1134)

File: skins/oasis/modules/OasisController.class.php [hunk 14]
```diff
@@ -292,28 +371,48 @@ private function loadJs() {
...
+
+		// generate direct script tags
+		foreach ($assets as $url) {
+			$url = htmlspecialchars( $url );
+			$jsLoader .= "<script src=\"{$url}\"></script>\n";
+		}
+
+		$tpl = $this->app->getSkinTemplateObj();
+
+		// $tpl->set( 'headscripts', $out->getHeadScripts() . $out->getHeadItems() );
```
Comment (WikiaTech) [inline skins/oasis/modules/OasisController.class.php:399]: MAJOR Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)

File: extensions/wikia/WikiaPhotoGallery/WikiaPhotoGalleryHelper.class.php [hunk 15]
```diff
@@ -94,8 +94,8 @@ static public function setupEditPage($editform) {
 
 		wfProfileIn( __METHOD__ );
 
-		if ( $app->checkSkin( [ 'oasis' ] ) ) {
-			$app->wg->Out->addModules( 'ext.WikiaPhotoGallery' );
+		if ( F::app()->checkSkin( [ 'oasis' ] ) ) {
+			$app->wg->Out->addScript("<script type=\"{$app->wg->JsMimeType}\" src=\"{$app->wg->ExtensionsPath}/wikia/WikiaPhotoGallery/js/WikiaPhotoGallery.js\"></script>\n");
 
 			// load message for MW toolbar button tooltip
 			$wgHooks['MakeGlobalVariablesScript'][] = 'WikiaPhotoGalleryHelper::makeGlobalVariablesScript';
```
Comment (WikiaTech) [inline extensions/wikia/WikiaPhotoGallery/WikiaPhotoGalleryHelper.class.php:101]: MAJOR Remove this unused "$wgHooks" local variable. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 118 issues
- MAJOR 61 major
- MINOR 47 minor
- INFO 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [FacebookClientHooks.class.php#L56](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/FacebookClient/FacebookClientHooks.class.php#L56): Remove the unused function parameter "$skin". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
2. MAJOR [FacebookClientHooks.class.php#L130](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/FacebookClient/FacebookClientHooks.class.php#L130): Remove the unused function parameter "$html". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
3. MAJOR [FacebookClientHooks.class.php#L176](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/FacebookClient/FacebookClientHooks.class.php#L176): Remove the unused function parameter "$user". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
4. MAJOR [SpecialSiteWideMessages.php#L41](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php#L41): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
5. MAJOR [SpecialSiteWideMessages.php#L79](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php#L79): Remove the unused function parameter "$skin". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
6. MAJOR [SpecialSiteWideMessages.php#L171](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php#L171): Remove the unused function parameter "$reason". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
7. MAJOR [SpecialSiteWideMessages.php#L180](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/SiteWideMessages/SpecialSiteWideMessages.php#L180): Remove the unused function parameters "$dbw, $uid, $oldusername, $newusername, $process". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
8. MAJOR [WikiaPhotoGalleryHelper.class.php](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/WikiaPhotoGallery/WikiaPhotoGalleryHelper.class.php): File "WikiaPhotoGalleryHelper.class.php" has 1315 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS104)
9. MAJOR [WikiaPhotoGalleryHelper.class.php#L3](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/WikiaPhotoGallery/WikiaPhotoGalleryHelper.class.php#L3): Class "WikiaPhotoGalleryHelper" has 35 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
10. MAJOR [WikiaPhotoGalleryHelper.class.php#L75](https://github.com/Wikia/app/blob/b39c6926227f4309f2ff3396a099562ae9275fa1/extensions/wikia/WikiaPhotoGallery/WikiaPhotoGalleryHelper.class.php#L75): Remove the unused function parameters "$params, $frame". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

**Comment by kerriamber** [comment]
ping @mixth-sense

**Comment by ghost** [comment]
LGTM :+1:

**Pull request merged**